### PR TITLE
755 Feature request add senders id to the chat reaction information re-opened

### PR DESCRIPTION
**Pull request by CapoMK25**
### Brief description

The UI-team lead reached out and told me about them seeing that the chat was sending messages in two different formats to the player, this as the first message: 

`{"message":{"chat":"clan","event":"newReaction","message":{"_id":"69147917d014d2f595a278c4","clan_id":"6740af56d977418ddbe08e29","content":"Rationaalinen analyysini tilanteesta tukee johtopäätöstä, että mukana oleminen kannattaa. Liityn siis seuraan. ","type":"clan","feeling":"Love","reactions":[{"playerName":"BillTheBeast","emoji":"Sad"},{"playerName":"BTB3","emoji":"Sad","sender_id":"69066ef167be4c41fdcde73a"}],"createdAt":"2025-11-12T12:09:59.580Z"}}}`

And this as the second: 

`[{"_id":"69147917d014d2f595a278c4","type":"clan","sender_id":"69066ef167be4c41fdcde73a","feeling":"Love","content":"Rationaalinen analyysini tilanteesta tukee johtopäätöstä, että mukana oleminen kannattaa. Liityn siis seuraan. ","clan_id":"6740af56d977418ddbe08e29","reactions":[{"playerName":"BillTheBeast","emoji":"Sad"},{"playerName":"BTB3","emoji":"Sad","sender_id":"69066ef167be4c41fdcde73a"}],"createdAt":"2025-11-12T12:09:59.580Z","updatedAt":"2026-02-11T12:42:56.361Z","__v":0,"id":"69147917d014d2f595a278c4"},null]`

After a thorough investigation I think I found the culprit and decided to re-open this PR here. Essentially the UI side is dealing with a case of duplicate messages if this change isn't implemented. 

### Change list

- Fixed Array Serialization Issue: Resolved the "trailing null" bug in the chat history retrieval logic. The service now correctly filters out null entries and ensures a clean JSON array structure, preventing client-side crashes in Unity.

Reaction Schema Standardization: Updated the `ChatMessage` schema to include a mandatory `sender_id` field within the reactions array.

Legacy Data Support: Implemented a default value of a "legacy_system" for `sender_id` in reactions to maintain backwards compatibility with messages generated prior to this overhaul.

Query Logic Refinement: Identified and addressed a type-mismatch where `clan_id` was inconsistently handled as a String versus an `ObjectId`. Ensured that history queries now reliably match based on the player's active `clan_id`.

Unit Test Validation: Updated the `chat.service.spec.ts` to include edge cases for empty emojis and reaction updates, ensuring the logic handles removals without corrupting the message object.

Technical Notes

[!IMPORTANT] During local testing, it was discovered that the `PATCH /chat` endpoint is currently restricted to `TESTING_SESSION` environments and requires `IsGroupAdmin` privileges. For manual verification of the reaction logic, use the provided MongoDB shell scripts to simulate legacy data scenarios.

Manual Verification Results

Verified via Postman:

Endpoint: `GET /chat/history?type=clan&recipientId={{clan_id}}`

Result: 200 OK.

Structure: Confirmed flat array format without the trailing null object.

Data Integrity: Reactions now consistently present the `sender_id` field as expected by the UI team.

### CODE REVIEW

File: src/chat/service/chat.service.ts [hunk 1]
```diff
@@ -54,34 +54,36 @@ export class ChatService {
...
+    update,
+    { new: true, ...options }
+  ).exec();
 
-    return [message, null];
+  if (!updatedDoc) {
+    return [null, [new ServiceError({ reason: SEReason.NOT_FOUND, message: 'Msg not found' })]];
   }
 
+  return [updatedDoc as unknown as ChatMessageDto, null];
```
Comment (CapoMK25) [inline src/chat/service/chat.service.ts]: Fixed a trailing null issue in history that was causing Unity client crashes for the UI team. Verified via Postman history logs.

**Review by hoan301298 — COMMENTED** [review]
I think we should have some suggestions from UI Team how to test with your solutions by using WebSocket. This is required since the bug was coming from sending messages during WebSocket Event (`@SubscribeMessage()`).

**Review by hoan301298 — APPROVED** [review]
This issue caused from last PR with adding transaction methods to `ChatModule`. Since when transaction is failed or not, it will return with/without `errors` or `newMessage`.

To fix this issue: remove `return` at the end of the handlers

Remove transaction support for create-new-message-event due to there is only one action with database and in this case, transaction does not help!

The solution has been tested via WebSocket in Postman that only message has been returned.

This PR is now ready to merge into `dev` branch

Image
